packages/guides/src/Compiler/ShadowTree/TreeNode.php

```diff
@@ -123,6 +124,18 @@ public function addChild(Node $child): void
         $this->node->addChildNode($child);
     }
 
+    public function pushChild(Node $child): void
+    {
+        if ($this->node instanceof CompoundNode === false) {
```

Comment:
No need for strict comparison, we know this is a boolean

```suggestion
        if (! $this->node instanceof CompoundNode) {
```

Reply:
The exclamation mark might be missed indeed.